prowler/providers/azure/services/containerregistry/containerregistry_service.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+
+from azure.mgmt.containerregistry import ContainerRegistryManagementClient
+
+from prowler.lib.logger import logger
+from prowler.providers.azure.azure_provider import AzureProvider
+from prowler.providers.azure.lib.service.service import AzureService
+from prowler.providers.azure.services.monitor.monitor_client import monitor_client
+from prowler.providers.azure.services.monitor.monitor_service import DiagnosticSetting
+
+
+########################## Container Registry
+class ContainerRegistry(AzureService):
+    def __init__(self, provider: AzureProvider):
+        super().__init__(ContainerRegistryManagementClient, provider)
```

Comment:
I think you are not using the provider in the method

```suggestion
        super().__init__(ContainerRegistryManagementClient)
```

Reply:
It seems its an mandatory argument for the **AzureService** class:

```
2024-09-27 15:50:04,852 [File: check.py:653]    [Module: check]  ERROR: TypeError[582]: AzureService.__init__() missing 1 required positional argument: 'provider'
```

So I wouldnt want to change it.